go/ephemeral/device_ek_storage.go

```diff
@@ -189,3 +212,55 @@ func (s *DeviceEKStorage) MaxGeneration(ctx context.Context) (maxGeneration keyb
 	}
 	return maxGeneration, nil
 }
+
+func (s *DeviceEKStorage) DeleteExpired(ctx context.Context) (expired []keybase1.EkGeneration, err error) {
+	defer s.G().CTrace(ctx, "DeviceEKStorage#DeleteExpired", func() error { return err })()
+	s.Lock()
+	defer s.Unlock()
+
+	cache, err := s.getCache(ctx)
+	if err != nil {
+		return expired, err
```

Comment:
You could have all the error cases `return nil, err` here, since `nil` is a valid slice.